core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -230,6 +228,61 @@ oppia.factory('TopicUpdateService', [
           // Get the current change list.
           var currentChangeList = UndoRedoService.getChangeList();
           var indicesToDelete = [];
+          var oldSubtopicIdSkillIdMap = {};
+          var index = null;
+          var skillIds = subtopic.getSkillIds();
+          for (var i = 0; i < currentChangeList.length; i++) {
+            var changeDict =
+              currentChangeList[i].getBackendChangeObject();
+            if (oldSubtopicIdSkillIdMap[changeDict.skill_id] === undefined) {
```

Comment:
Shouldn't you be doing some kind of filtering on the _type_ of change first?

Reply:
Actually, only move commands have a new_subtopic_id field, so I didn't have that check. If such a field doesn't exist, then oldSubtopicIdSkillMap is not set and therefore no operation happens. I could add a check, though I am not sure if new_subtopic_id will be used in any change command.

Reply:
Done, added the check.